include/SDL_events.h

```diff
@@ -594,12 +594,14 @@ typedef struct SDL_OSEvent
  */
 typedef struct SDL_UserEvent
 {
-    Uint32 type;        /**< ::SDL_USEREVENT through ::SDL_LASTEVENT-1 */
-    Uint32 timestamp;   /**< In milliseconds, populated using SDL_GetTicks() */
-    Uint32 windowID;    /**< The associated window if any */
-    Sint32 code;        /**< User defined event code */
-    void *data1;        /**< User defined data pointer */
-    void *data2;        /**< User defined data pointer */
+    Uint32 type;                    /**< ::SDL_USEREVENT through ::SDL_LASTEVENT-1 */
+    Uint32 timestamp;               /**< In milliseconds, populated using SDL_GetTicks() */
+    Uint32 windowID;                /**< The associated window if any */
+    Sint32 code;                    /**< User defined event code */
+    void *data1;                    /**< User defined data pointer */
+    void *data2;                    /**< User defined data pointer */
+    void (*free)(void *, void *);   /**< User specified function for freeing user defined 
```

Comment:
Thanks. Will push the fix shortly

Reply:
It would be also good to rebase to current main branch, squash the commits into a single one and then force-push, so that we have a clean git history if this gets accepted.